### PR TITLE
Simplify vision ops

### PR DESCRIPTION
**Pull request by RandySheriffH**
Simplify vision op by utilizing latest custom ops APIs.

### CODE REVIEW

File: operators/vision/decode_image.cc [hunk 1]
```diff
@@ -37,8 +35,41 @@ void KernelDecodeImage::Compute(OrtKernelContext* context) {
...
+//  // Setup output & copy to destination
+//  const cv::Size decoded_image_size = decoded_image.size();
+//  const int64_t height = decoded_image_size.height;
+//  const int64_t width = decoded_image_size.width;
+//  const int64_t colors = decoded_image.elemSize();  //  == 3 as it's BGR
+//
+//  const std::vector<int64_t> output_dims{height, width, colors};
+//  OrtValue* output_value = ort_.KernelContext_GetOutput(context, 0, output_dims.data(), output_dims.size());
+//  uint8_t* decoded_image_data = ort_.GetTensorMutableData<uint8_t>(output_value);
+//  memcpy(decoded_image_data, decoded_image.data, narrow<size_t>(height * width * colors));
```
Comment (RandySheriffH) [inline operators/vision/decode_image.cc:73]: Will be removed.

**Comment by wejoncy** [comment]
Would the new API support multi-thread?

**Comment by skottmckay** [comment]
> Thanks for the Simplifying work. Would the new API support multi-thread?

I believe the ORT Custom Op C++ API is just a wrapper over the C API. Same things are supported but should be simpler/easier to use the C++ API.

**Review by skottmckay — APPROVED** [review]
:shipit:

**Comment by RandySheriffH** [comment]
> Thanks for the Simplifying work. Would the new API support multi-thread?

Not yet, but we will be there.

**Pull request merged**